test/test_rpc.py

```
@@ -0,0 +1,170 @@
+        opts.devices = [dist.ProcessGroupGloo.create_tcp_device(interface="lo")]
+        opts.timeout = 5.0
+        opts.threads = threads
+        return opts
+
+    @classmethod
+    def _init_rpc(cls, rank, filename, world_size):
+        store = dist.FileStore(filename, world_size)
+        dist.init_process_group(
+            backend='gloo', rank=rank, world_size=world_size, store=store)
+        dist.init_rpc('worker%d' % rank)
+
+    @classmethod
+    def _destroy_rpc(cls):
+        dist.destroy_rpc()
+        dist.destroy_process_group(dist.group.WORLD)
+
+    # Why classmethod? multiprocessing cannot pickle TestCase subclass when in
+    # spawn mode. See https://bugs.python.org/issue33884.
+    @classmethod
+    def _test_process(cls, rank, func, filename, c2p, p2c, world_size):
+        RpcTest._init_rpc(rank, filename, world_size)
+
+        result, expected = func(rank, world_size)
+        c2p.put((rank, expected, result))
+        p2c.get()
+
+        RpcTest._destroy_rpc()
+
+    def _test_multiprocess(self, func, world_size):
+        # file store will delete the test file on destruction
+        file = tempfile.NamedTemporaryFile(delete=False)
+        ctx = mp.get_context('spawn')
+        c2p = ctx.Queue(10)
+        p2c = ctx.Queue(10)
+
+        ps = []
+        for i in range(world_size):
+            p = ctx.Process(
+                target=RpcTest._test_process,
+                args=(i, func, file.name, c2p, p2c, world_size))
+            p.start()
+            ps.append(p)
+
+        for _ in range(world_size):
+            rank, expected, result = c2p.get()
+            self.assertEqual(
+                expected,
+                result,
+                (
+                    "Expect rank {} to receive tensor {} but got {}."
+                ).format(rank, expected, result)
+            )
+
+        for _ in range(world_size):
+            p2c.put(0)
+
+        for p in ps:
+            p.join(2)
```

This is masking errors in child process.

If a child process crashes, parent could still join it without any error.

Can you add exitcode check here, for example, `assert p.exitcode ==0` ?

---

I will use the `MultiProcessTestCase` for testing after #23660 gets merged. That should address the problem.